### ISSUE TITLE
False Positive for `Lint/ShadowedArgument`

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#13227]
## Expected behavior

Assigning to a local variable shadowing an argument, and then calling `super` is a useful thing to do if you're trying to modify the argument being passed to `super`. (The cop still makes sense if there is no `super` call).

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

## Steps to reproduce the problem

test.rb:
```ruby
#!/usr/bin/env ruby
# frozen_string_literal: true

# Superclass
class A
  def a(options = nil)
    puts options.inspect
  end
end

# Subclass
class B < A
  def a(namespace: nil, **)
    namespace = nil
    super
  end
end

B.new.a
B.new.a(namespace: 'bob')
```

```sh
% rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:14:5: W: Lint/ShadowedArgument: Argument namespace was shadowed by a local variable before it was used.
    namespace = nil
    ^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

Note the output of running test.rb:
```sh
% ./test.rb
{:namespace=>nil}
{:namespace=>nil}
```

In the second call where `namespace` is passed, the argument is meaningfully changed by `B#a` before the call to `A#a`.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
% rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.2, running on ruby 3.3.3) [arm64-darwin23]
```

**Comment by Earlopain** [comment]
You can set `IgnoreImplicitReferences` to true for this to not be a offense: https://docs.rubocop.org/rubocop/cops_lint.html#ignoreimplicitreferences-true

**Comment by ccutrer** [comment]
Ahhh, thanks!
